modules/transformer-batch/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/batch/ShredJob.scala

```diff
@@ -170,6 +170,12 @@ class TypeAccumJob(@transient val spark: SparkSession,
         case Left(_) =>
           List.empty
       }.distinct().collect().toList
+
+    config.deduplication.synthetic match {
+      case TransformerConfig.Deduplication.Synthetic.None => types
+      // Include duplicate schema to types list if synthetic deduplication is enabled
+      case _ => TypesInfo.WideRow.Type(Deduplication.DuplicateSchema, SnowplowEntity.Context) :: types
```

Comment:
Is there any chance that `types` can already include the duplicate schema?  I know it's unlikely, and it probably means someone sent a very weird event.

But.....

Would anything break if that happens?  We must not leave a vulnerability where someone can send a weird event and then it breaks loading.

Reply:
I don't think it would be included in any event in normal scenario.

Even it included explicitly, it would be just another context in the event. It wouldn't create problem for loader.

Reply:
> I don't think it would be included in any event in normal scenario.

Agreed, not a _normal scenario_.  But think of it like a hacking attempt.

> Even it included explicitly, it would be just another context in the event. It wouldn't create problem for loader.

What's worrying me, is that it's possible for the same schema to be in the list twice.  So to rephrase what I'm asking: what happens when the same schema (any schema!) is in the list twice.  Is that technically OK?

Reply:
I think it should be okay. Even it is expected in some cases I think. This is the reason why we are making column type of contexts `JsonArray` [in Snowflake Loader](https://github.com/snowplow/snowplow-rdb-loader/blob/master/modules/snowflake-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/snowflake/Snowflake.scala#L60).